File: MFDLabs.Grid.Bot/Commands/CloseJob.cs
using System.Linq;
using System.Threading.Tasks;
using Discord.WebSocket;
using MFDLabs.Grid.Bot.Extensions;
using MFDLabs.Grid.Bot.Interfaces;
using MFDLabs.Grid.Bot.Utility;

namespace MFDLabs.Grid.Bot.Commands
{
    internal class CloseJob : IStateSpecificCommandHandler
    {
        public string CommandName => "Close Grid Server Job";
        public string CommandDescription => $"Attempts to close a grid server job via the SoapUtility\nLayout: {MFDLabs.Grid.Bot.Properties.Settings.Default.Prefix}closejob jobID.";
        public string[] CommandAliases => new string[] { "cj", "closejob" };
        public bool Internal => true;
        public bool IsEnabled { get; set; } = true;

        public async Task Invoke(string[] messageContentArray, SocketMessage message, string originalCommand)
        {
            if (!await message.RejectIfNotAdminAsync()) return;

            var jobID = messageContentArray.ElementAtOrDefault(0);

            if (jobID == default)
            {
                await message.ReplyAsync($"Missing required parameter 'jobId', the layout is: {MFDLabs.Grid.Bot.Properties.Settings.Default.Prefix}{originalCommand} jobID");
                return;
            }

            await GridServerArbiter.Singleton.CloseJobAsync(jobID);
            await message.ReplyAsync($"Successfully closed grid server jobs '{jobID}'.");
        }
    }
}

File: Source/Lib/Commands/CloseJob.ts
import { Message } from 'discord.js';
import { GlobalConfig } from '../Utility/GlobalConfig';
import { AdminUtil } from '../Administration/AdminUtil';
import { SOAPUtil } from '../RCCService/SOAPUtility';
import { StateSpecificCommandHandler } from '../CommandUtility/Types/StateSpecificCommandHandler';

class CloseJob implements StateSpecificCommandHandler {
	public CommandName = 'Close Job';
	public CommandDescription = 'Closes a job with the parameters of jobIDD';
	public Internal = true;
	public Command = ['cj', 'closejob', 'closegame'];
	public Callback = async (messageContent: string[], messageRoot: Message) => {
		if (!AdminUtil.RejectIfNotAdmin(messageRoot)) return;

		const jobId = messageContent[0];

		if (!jobId) return messageRoot.reply(`Missing required parameter 'jobId', the layout is: ${GlobalConfig.Prefix}closejob jobID`);

		await SOAPUtil.CloseJob(jobId);
		return messageRoot.reply(`Successfully closed the job '${jobId}'`);
	};
}

export = new CloseJob();
